pandas/core/arrays/timedeltas.py

```diff
@@ -162,15 +162,30 @@ def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
         result._freq = freq
         return result
 
-    def __new__(cls, values, freq=None, dtype=_TD_DTYPE):
+    def __new__(cls, values, freq=None, dtype=_TD_DTYPE, copy=False):
 
         freq, freq_infer = dtl.maybe_infer_freq(freq)
 
-        values = np.array(values, copy=False)
-        if values.dtype == np.object_:
-            values = array_to_timedelta64(values)
+        values, inferred_freq = sequence_to_td64ns(
+            values, copy=copy, unit=None)
+        if inferred_freq is not None:
+            if freq is not None and freq != inferred_freq:
+                raise ValueError('Inferred frequency {inferred} from passed '
+                                 'values does not conform to passed frequency '
+                                 '{passed}'
+                                 .format(inferred=inferred_freq,
+                                         passed=freq.freqstr))
+            elif freq is None:
+                # TODO: should this be the stronger condition `if freq_infer`?
```

Comment:
Can you clarify this TODO comment?

Reply:
Sure.  We might also be able to resolve the answer as "no" in this thread.

ATM if we get an `inferred_freq` here, we are setting `freq = inferred_freq` regardless of whether the user passed `freq="infer"`.  In particular, we are ignoring the possibility that the user passed `freq=None` specifically wanting the result to have no freq.